### PR TITLE
Prometheus: Display HELP and TYPE of metrics if available

### PR DESCRIPTION
**Pull request by davkal**
- Prometheus recently added a metadata API around HELP and TYPE of
metrics, see https://github.com/prometheus/prometheus/pull/6442
- request metadata when datasource instance is created
- use metadata to show help and type in typeahead suggestions and in
metrics selector as tooltip

Screenshot 2019-12-16 at 17 45 53

### CODE REVIEW

File: public/app/plugins/datasource/prometheus/language_provider.ts [hunk 1]
```diff
@@ -85,7 +95,7 @@ export default class PromQlLanguageProvider extends LanguageProvider {
     return PromqlSyntax;
   }
 
-  request = async (url: string) => {
+  request = async (url: string, defaultValue: any): Promise<any> => {
```
Comment (davkal) [inline public/app/plugins/datasource/prometheus/language_provider.ts]: The response for the new API is a dict, as opposed to a list for all other metadata queries. Had to introduce a mandatory default value to make sure caller is aware of return type by having to give a default value.

File: public/app/plugins/datasource/prometheus/language_provider.ts [hunk 2]
```diff
@@ -95,12 +105,13 @@ export default class PromQlLanguageProvider extends LanguageProvider {
...
 
-    return [];
+    return defaultValue;
   };
 
   start = async (): Promise<any[]> => {
-    this.metrics = await this.request('/api/v1/label/__name__/values');
+    this.metrics = await this.request('/api/v1/label/__name__/values', []);
     this.lookupsDisabled = this.metrics.length > this.lookupMetricsThreshold;
+    this.metricsMetadata = await this.request('/api/v1/metadata', {});
```
Comment (davkal) [inline public/app/plugins/datasource/prometheus/language_provider.ts]: Could make this conditional on `lookupsDisabled`, but the responses seems small and fast enough.

File: packages/grafana-ui/src/components/Cascader/Cascader.tsx [hunk 3]
```diff
@@ -9,6 +9,8 @@ export interface CascaderOption {
 
   children?: CascaderOption[];
   disabled?: boolean;
+  // Undocumented tooltip API
```
Comment (davkal) [inline packages/grafana-ui/src/components/Cascader/Cascader.tsx]: https://github.com/react-component/cascader/blob/7d4e8aae7136b8fc8eea1ab55d05461f4d78c908/src/Menus.tsx#L101

**Review by Estrax — APPROVED** [review]
LGTM

File: public/app/plugins/datasource/prometheus/language_provider.ts (hunk 1 above)
Comment (aocenas) [reply public/app/plugins/datasource/prometheus/language_provider.ts]: We could make it
```
request = async <T = any>(url: string, defaultValue: T): Promise<T> => {
```
To make it easier to add optional type safety.

**Pull request merged**